### PR TITLE
Dynamically generate robots.txt

### PR DESCRIPTION
**Pull request by jasonharrison**
Dynamically generate robots.txt

### CODE REVIEW

File: src/server.tsx [hunk 1]
```diff
@@ -104,4 +102,22 @@ const startServer = async () => {
   });
 };
 
+function getOpenRobotsTxt() {
+  const robotstxt = ['User-agent: *'];
```
Comment (flamecoals) [inline src/server.tsx]: please use consistent variable naming. In this case, robotTxt

File: src/server.tsx [hunk 2]
```diff
@@ -104,4 +102,22 @@ const startServer = async () => {
...
+  const AI = GameMode.AI;
+  const OnlineFriend = GameMode.OnlineFriend;
+  GAMES_LIST.forEach((game) => {
+    game.modes.forEach((modeInfo) => {
+      if (modeInfo.mode === GameMode.AI) {
+        robotstxt.push(`Disallow: /g/${game.code}/ai/`);
```
Comment (flamecoals) [inline src/server.tsx]: isnt AI supposed to be capital letters here?

File: src/server.tsx [hunk 3]
```diff
@@ -104,4 +102,22 @@ const startServer = async () => {
...
+      }
+      if (modeInfo.mode === GameMode.OnlineFriend) {
+        robotstxt.push(`Disallow: /g/${game.code}/online/`);
```
Comment (flamecoals) [inline src/server.tsx]: arent we supposed to have a * in the end?

File: src/server.tsx (hunk 2 above)
Comment (jasonharrison) [reply src/server.tsx]: Yes, you're right -- and robots.txt **is** case sensitive!  Thank you for catching!

File: src/server.tsx (hunk 3 above)
Comment (jasonharrison) [reply src/server.tsx]: Apparently some robots.txt parsers do not support wildcard syntax.  /online/ and such should work and disallow all subdirectories.

**Pull request merged**